include/aws/cal/private/rsa.h

```diff
@@ -0,0 +1,115 @@
+#ifndef AWS_C_CAL_PRIVATE_RSA_H
+#define AWS_C_CAL_PRIVATE_RSA_H
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#include <aws/cal/rsa.h>
+
+#include <aws/common/byte_buf.h>
+#include <aws/common/ref_count.h>
+
+struct aws_rsa_key_pair;
+struct aws_der_decoder;
+
+struct aws_rsa_key_vtable {
+    void (*destroy)(struct aws_rsa_key_pair *key_pair);
+
+    int (*encrypt)(
+        struct aws_rsa_key_pair *key_pair,
+        enum aws_rsa_encryption_algorithm algorithm,
+        struct aws_byte_cursor plaintext,
+        struct aws_byte_buf *out);
+    int (*decrypt)(
+        struct aws_rsa_key_pair *key_pair,
+        enum aws_rsa_encryption_algorithm algorithm,
+        struct aws_byte_cursor ciphertext,
+        struct aws_byte_buf *out);
+
+    int (*sign)(
+        struct aws_rsa_key_pair *key_pair,
+        enum aws_rsa_signing_algorithm algorithm,
+        struct aws_byte_cursor digest,
+        struct aws_byte_buf *out);
+
+    int (*verify)(
+        struct aws_rsa_key_pair *key_pair,
+        enum aws_rsa_signing_algorithm algorithm,
+        struct aws_byte_cursor digest,
+        struct aws_byte_cursor signature);
+};
+
+struct aws_rsa_key_pair {
+    struct aws_allocator *allocator;
+    struct aws_rsa_key_vtable *vtable;
+    struct aws_ref_count ref_count;
+
+    size_t key_size_in_bits;
+    struct aws_byte_buf priv;
+    struct aws_byte_buf pub;
+
+    bool good;
+    void *impl;
+};
+
+void aws_rsa_key_pair_destroy(void *key_pair);
+
+/*
+ * RSAPrivateKey as defined in RFC 8017 (aka PKCS1 format):
+ *   version           Version,
+ *   modulus           INTEGER,  -- n
+ *   publicExponent    INTEGER,  -- e
+ *   privateExponent   INTEGER,  -- d
+ *   prime1            INTEGER,  -- p
+ *   prime2            INTEGER,  -- q
+ *   exponent1         INTEGER,  -- d mod (p-1)
+ *   exponent2         INTEGER,  -- d mod (q-1)
+ *   coefficient       INTEGER,  -- (inverse of q) mod p
+ *   otherPrimeInfos   OtherPrimeInfos OPTIONAL
+ *   Note: otherPrimeInfos is used for >2 primes RSA cases, which are not very
+ *   common and currently not supported by CRT. Version == 0 indicates 2 prime
+ *   case and version == 1 indicates >2 prime case, hence in practice it will
+ *   always be 0.
+ */
+struct s_rsa_private_key_pkcs1 {
```

Comment:
trivial: the `s_` prefix is for static functions, not structs. yes it's kinda arbitrary, but that's how we've been doing it
```suggestion
struct aws_rsa_private_key_pkcs1 {
```

(comment applies to several structs in this PR)